Simplenote/Classes/NotesListController.swift

```diff
@@ -0,0 +1,216 @@
+import Foundation
+
+
+// MARK: - NotesListController
+//
+class NotesListController: NSObject {
+
+    /// Main Context
+    ///
+    private let viewContext: NSManagedObjectContext
+
+    /// Notes Controller
+    ///
+    private lazy var notesController: ResultsController<Note> = {
+        ResultsController<Note>(viewContext: viewContext, matching: state.predicateForNotes(filter: filter), sortedBy: sortMode.descriptorsForNotes)
+    }()
```

Comment:
what's going on with the function wrapper? this is lazy by nature of the keyword, right, not by nature of the `{}()`? in fact, does this do anything?

Reply:
It's a lazily initialized property. Nil until the first time it's accessed, _then_ it'll contain a ResultsController for the Note entity.

It reads as `we want an ivar named notesController, of the ResultsController kind. Please initialize it with the result of this closure`.

Another detail is: in a closure / method, if there's a single line, the language assumes it's the return value. No need to write `return` anymore. 

🔥

Reply:
sure, but wouldn't this be the same as the following code?

```swift
private lazy var notesController: ResultsController<Note> = \
    ResultsController<Note>(viewContext: viewContext, matching: state.predicateForNotes(filter: filter), sortedBy: sortMode.descriptorsForNotes)
```

Reply:
Let's settle on the following, since it's more swift-y!

```
    private lazy var tagsController = ResultsController<Tag>(viewContext: viewContext,
                                                             matching: state.predicateForTags(),
                                                             sortedBy: sortMode.descriptorsForTags)
```

Adding semicolons in a new line feels .. way too JS!

Thank you!